corehq/motech/fhir/utils.py

```diff
@@ -92,3 +94,15 @@ def build_capability_statement(domain, fhir_version):
         "fhirVersion": fhir_version,
         "date": CAPABILITY_STATEMENT_PUBLISHED_DATE,
     }
+
+
+def case_access_authorized(domain, access_token, case_id):
+    """Case Access is allowed if:
+    - There exists a CaseRelationship for this access token
+    - There exist a CaseRelationship for any ancestor cases for this access token
```

Comment:
Is there a logical reason for this rule? It seems like it is a totally differerent permission model than mobile, which I find a bit odd.

Reply:
This is a great question, and maybe worth more discussion than a github comment... 

The permissions model here is very different to mobile. The user in this case is the person who is described by the "demographic case" - the case they have assigned to them through the Consumer User invitation workflow. On mobile it is based on the owner_id, which has no bearing to the consumer user at all.

We don't currently have a good way of saying "this case is about the same person" other than through case relationships. We had decided that by using the case hierarchy tree, we could be reasonably confident that a case residing _below_ another case was about that same top-level person. It's true that this doesn't always hold, like when you have child cases that model actual children of the same case type as the parent. We had talked about preventing that situation somehow through search code in the API methods, but I don't think that got implemented in this round. I'm adding it to our backlog of future work for this feature.

The other option here is to make all of these checks based on a list, and have the invitation workflow link a Consumer User with a list of cases that they should be able to share with a third-party system. The downside of this is that all cases need to already exist when the invitation is sent, and any newly created cases will have to be added to a Consumer User Case Relationship manually. 

It's definitely a trade-off between usability and permissiveness, and for the POC I went with usability. 

Maybe you have some other thoughts on how we could use the current mobile permissions model that I didn't think about?

Reply:
> The other option here is to make all of these checks based on a list, and have the invitation workflow link a Consumer User with a list of cases that they should be able to share with a third-party system

Do the child cases also get made "available" to the consumer user? I imagine the big, scary thing is like, assigning a "facility" case someone, and then they can view every patient child case in the facility (contrived example, but you get it).

If there's also a check ensuring the other cases should be available to the consumer user, then I imagine automatically serving them with the parent is fine. But I didn't see that check. And I realize this also makes the code around the creation of the consumer user case mappings more complex.

> for the POC I went with usability

I agree with this decision.

> Maybe you have some other thoughts on how we could use the current mobile permissions model that I didn't think about?

The first thought I have on this is something that should be familiar to you: extension cases! But I recognize that if people are building this into their own apps then that's severely limiting. But I think it *is* the exact same thing from a data-modeling perspective.

I'm not requesting any changes on this point so long as we feel we've appropriately weighed the risks of the chosen approach.